src/internal_events/statsd.rs

```diff
@@ -93,3 +96,29 @@ impl<T: std::fmt::Debug + std::fmt::Display> InternalEvent for StatsdSocketError
         );
     }
 }
+
+#[derive(Debug)]
+pub struct StatsdInvalidMetric<'a> {
+    pub value: &'a MetricValue,
+    pub kind: &'a MetricKind,
+}
+
+impl<'a> InternalEvent for StatsdInvalidMetric<'a> {
+    fn emit_logs(&self) {
+        warn!(
+            message = "Invalid metric sent; dropping event.",
+            value = ?self.value,
+            kind = ?self.kind,
+            rate_limit_secs = 30,
+        )
+    }
+
+    fn emit_metrics(&self) {
+        counter!(
+            "processing_errors", 1,
+            "component_kind" => "sink",
+            "component_type" => "statsd_metrics",
```

Comment:
I think this should be just "statsd" instead of "statsd_metrics". 

https://github.com/timberio/vector/blob/c9317c5893ac87f3ba1845ad23908b7ef102e5df/src/sinks/statsd.rs#L78-L80